flexmeasures/data/schemas/generic_assets.py

```diff
@@ -42,32 +42,24 @@ class GenericAssetSchema(ma.SQLAlchemySchema):
     longitude = LongitudeField(allow_none=True)
     generic_asset_type_id = fields.Integer(required=True)
     attributes = JSON(required=False)
+    parent_asset_id = fields.Int(required=False, allow_none=True)
 
     class Meta:
         model = GenericAsset
 
     @validates_schema(skip_on_field_errors=False)
-    def validate_name_is_unique_in_account(self, data, **kwargs):
-        if "name" in data:
-            if data.get("account_id") is None:
-                asset = GenericAsset.query.filter(
-                    GenericAsset.name == data["name"], GenericAsset.account_id.is_(None)
-                ).first()
-                if asset:
-                    raise ValidationError(
-                        f"A public asset with the name {data['name']} already exists.",
-                        "name",
-                    )
-            else:
-                asset = GenericAsset.query.filter(
-                    GenericAsset.name == data["name"],
-                    GenericAsset.account_id == data["account_id"],
-                ).first()
-                if asset:
-                    raise ValidationError(
-                        f"An asset with the name {data['name']} already exists in this account.",
-                        "name",
-                    )
+    def validate_name_is_unique_under_parent(self, data, **kwargs):
+        if "name" in data and data.get("parent_asset_id") is not None:
```

Comment:
I'm wondering what should happen to assets without a parent (such as those at the top level of a hierarchy, or orphaned assets). Even if we don't have a db constraint, should we still discourage users to name such assets the same? I'm probably overthinking. Just wanted to make note of it.

Reply:
Good point.

I've modified the code to check also for `parent_asset_id=None' which makes sure that the name is unique among the public assets.

Reply:
Actually, `parent_asset_id == None` does not filter public assets, it filters parentless assets, and without an `account_id` filter, it looks for them in *all* accounts. This would mean that a parentless asset in *any* account would lead to a global name reservation for asset edits passing through this schema.